eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/AbstractEventStorageEngine.java

```diff
@@ -106,20 +112,51 @@ public void storeSnapshot(DomainEventMessage<?> snapshot) {
      * @param failedEvent The EventMessage that could not be persisted
      */
     protected void handlePersistenceException(Exception exception, EventMessage<?> failedEvent) {
-        String eventDescription;
-        if (failedEvent instanceof DomainEventMessage<?>) {
-            DomainEventMessage<?> failedDomainEvent = (DomainEventMessage<?>) failedEvent;
-            eventDescription =
-                    format("An event for aggregate [%s] at sequence [%d]", failedDomainEvent.getAggregateIdentifier(),
-                           failedDomainEvent.getSequenceNumber());
-        } else {
-            eventDescription = format("An event with identifier [%s]", failedEvent.getIdentifier());
-        }
+        String eventDescription = buildExceptionMessage(failedEvent);
         if (persistenceExceptionResolver != null && persistenceExceptionResolver.isDuplicateKeyViolation(exception)) {
-            throw new ConcurrencyException(eventDescription + " was already inserted", exception);
+            if (isFirstDomainEvent(failedEvent)) {
+                throw new AggregateAlreadyExistsException(eventDescription, exception);
+            }
+            throw new ConcurrencyException(eventDescription, exception);
         } else {
-            throw new EventStoreException(eventDescription + " could not be persisted", exception);
+            throw new EventStoreException(eventDescription, exception);
+        }
+    }
+
+    /**
+     * Check whether or not this is the first event, which means we are trying to create the aggregate.
+     *
+     * @param failedEvent the event to be checked
+     * @return true in case of first event, false otherwise
+     */
+    private boolean isFirstDomainEvent(EventMessage failedEvent) {
+        if (failedEvent instanceof DomainEventMessage<?>) {
+            return ((DomainEventMessage) failedEvent).getSequenceNumber() == 0L;
+        }
+        return false;
+    }
+
+    /**
+     * Build an exception message based on an EventMessage.
+     *
+     * @param failedEvent
+     * @return the created exception message
+     */
+    private String buildExceptionMessage(EventMessage failedEvent) {
+        String eventDescription = format("An event with identifier [%s] could not be persisted",
+                                         failedEvent.getIdentifier());
+        if (isFirstDomainEvent(failedEvent)) {
+            DomainEventMessage failedDomainEvent = (DomainEventMessage) failedEvent;
+            eventDescription = format("An aggregate of type [%s] with id [%s] was already created.",
```

Comment:
```suggestion
            eventDescription = format("Cannot reuse aggregate identifier [%s] to create aggregate [%s] since identifiers need to be unique.",
```

This also means the parameters to the `format()` method should be switched by the way.